### PR TITLE
lisa.pelt: Fixup simulate_pelt() clock on task enqueue/dequeue

### PR DESCRIPTION
**Pull request by douglas-raillard-arm**
The PELT clock is typically not available on task enqueue/dequeue,
but aligning the activations on the preceding PELT events makes the
sleep time inaccurate.

Fix that by providing a fake clock sample based on the ftrace clock
delta, rescaled using the nearest time scaling factor.

### CODE REVIEW

File: lisa/tests/scheduler/load_tracking.py [hunk 1]
```diff
@@ -328,7 +328,21 @@ def get_simulated_pelt(self, task, signal_name):
             logger.warning('PELT clock is not available, ftrace timestamp will be used at the expense of accuracy')
             clock = None
 
-        df['simulated'] = simulate_pelt(df_activation['active'], index=df.index, init=init, clock=clock)
+        try:
+            capacity = trace.analysis.load_tracking.df_cpus_signal('capacity', cpus)
+        except MissingTraceEventError:
+            capacity = None
+        else:
+            cpu, = cpus
```
Comment (deggeman) [inline lisa/tests/scheduler/load_tracking.py:336]: Traceback (most recent call last):
  File "/opt/git/tools/lisa/tools/exekall/exekall/engine.py", line 2770, in genf
    val = self.callable_(**kwargs)
  File "/opt/git/tools/lisa/tools/exekall/exekall/engine.py", line 2354, in __call__
    return __UnboundMethod_self__.__wrapped__(*args, **kwargs)
  File "/opt/git/tools/lisa/lisa/tests/base.py", line 438, in wrapper
    res = func(self, *args, **kwargs)
  File "/opt/git/tools/lisa/lisa/trace.py", line 4659, in wrapper
    return f(self, *args, **kwargs)
  File "/opt/git/tools/lisa/lisa/tests/scheduler/load_tracking.py", line 667, in test_util_behaviour
    return self._test_all_freq(item_test)
  File "/opt/git/tools/lisa/lisa/tests/scheduler/load_tracking.py", line 688, in _test_all_freq
    item_res_bundles = [
  File "/opt/git/tools/lisa/lisa/tests/scheduler/load_tracking.py", line 689, in <listcomp>
    item_test(item)
  File "/opt/git/tools/lisa/lisa/tests/scheduler/load_tracking.py", line 664, in item_test
    return test_item.test_util_behaviour(
  File "/opt/git/tools/lisa/lisa/tests/base.py", line 438, in wrapper
    res = func(self, *args, **kwargs)
  File "/opt/git/tools/lisa/lisa/trace.py", line 4659, in wrapper
    return f(self, *args, **kwargs)
  File "/opt/git/tools/lisa/lisa/tests/base.py", line 589, in filter_wrapper
    res = wrapped_test(self, *args, **test_kwargs)
  File "/opt/git/tools/lisa/lisa/tests/scheduler/load_tracking.py", line 493, in test_util_behaviour
    return self._test_behaviour('util', error_margin_pct)
  File "/opt/git/tools/lisa/lisa/trace.py", line 4659, in wrapper
    return f(self, *args, **kwargs)
  File "/opt/git/tools/lisa/lisa/tests/scheduler/load_tracking.py", line 386, in _test_behaviour
    df = self.get_simulated_pelt(task, signal_name)
  File "/opt/git/tools/lisa/lisa/trace.py", line 4659, in wrapper
    return f(self, *args, **kwargs)
  File "/opt/git/tools/lisa/lisa/tests/scheduler/load_tracking.py", line 336, in get_simulated_pelt
    cpu, = cpus
ValueError: too many values to unpack (expected 1)

Comment (deggeman) [reply lisa/tests/scheduler/load_tracking.py:336]: (Pdb) c
[2021-03-10 11:56:25,899][lisa.tests.scheduler.load_tracking.InvarianceItem] INFO  Ignored PIDs for noise contribution: [19234:invar0-0], [0]
[2021-03-10 11:56:27,428][lisa.trace.Trace] WARNING  Events sched_load_avg_cpu, sched_load_avg_task, sched_load_cfs_rq, sched_load_se not found in the trace: /opt/work/lisa/results/20210310_11:43:39_a5a8e1fb730f42239764b8b94eb2a32e/Invariance:test_util_behaviour/7e7e350f2c294b789d75863af5f5ea82/lisa.tests.scheduler.load_tracking.Invariance.from_target/1/invar_0@775000/trace.dat
> /opt/git/tools/lisa/lisa/tests/scheduler/load_tracking.py(339)get_simulated_pelt()
-> cpu, = cpus
(Pdb) cpus
[0, 5]

Comment (douglas-raillard-arm) [reply lisa/tests/scheduler/load_tracking.py:336]: thanks, I'll have to use the cpu from the activation dataframe to pick a column for each row of cpu capacity

**Comment by douglas-raillard-arm** [comment]
@deggeman PR updated with a dynamic selection of the capacity (depending on the current CPU the task is on)

**Comment by douglas-raillard-arm** [comment]
Updated with some fixes around in simulate_pelt() in the new logic (handle NaN more correctly in the clock and capacity).

**Pull request merged**